R/biting_process.R

```diff
@@ -157,10 +156,15 @@ simulate_bites <- function(
       n_bites <- rpois(1, expected_bites)
       if (n_bites > 0) {
         bitten <- fast_weighted_sample(n_bites, lambda)
-        bitten_humans$insert(bitten)
+        if(parameters$parasite == "falciparum"){
+          bitten_humans$insert(bitten)
+          renderer$render('n_bitten', bitten_humans$size(), timestep)
+        }
         if(parameters$parasite == "vivax"){
-          # p.v must pass through the number of bites each individual receives
-          n_bites_each <- table(bitten)
+          # p.v must pass through the number of bites per person
+          bites_per_person <- tabulate(bitten, nbins = length(lambda))
+          bitten_humans <- individual::IntegerVariable$new(initial_values = bites_per_person)
```

Comment:
I see how this seems tidier, but I have a few concerns.
1. We now have a single variable `bitten_humans` that means different things for Pf and Pv and ...
2. that leads to a mismatch in the `calculate_infections()` function, which takes the argument `bitten_humans()` currently defined as a bitset of bitten humans.

Would it be clearer to:
1. Create the bitset of `bitten_humans` based on `n_bites` and `lambda` for both Pf and Pf and pass this through to `calculate_infections()` as an argument.
2. Additionally create a `IntegerVariable` for vivax called and storing `n_bites`. This can be accessed in `calculate_infections()` directly in that function as you're already passing `variables` as an argument.

Reply:
I've given this a go (and rebased) and I think it's ready for review.